comet/redis_async_locks.py

```diff
@@ -231,11 +246,15 @@ def __init__(self, lock, name):
         This will not do anything to tasks waiting on the variable. These
         should be cleaned up before calling this. It will also not close the
         underlying `Lock`, but does need to acquire it to close the condition
-        variable.
+        variable. Raises a LockError if it can't acquire the lock.
         """
-        async with self.lock as r:
-            await r.execute("del", "WAITING")
-            await r.execute("del", self.condname)
+        r = await self.lock.acquire(no_block=True)
```

Comment:
This is probably fine, but it does seem like it could lead to a messier shutdown. I liked the idea of waiting to acquire the lock, but you're right that we don't want to wait forever. Maybe we could instead switch the acquire to take a timeout argument (that just goes onto the blpop) and then set a timeout here for a couple of seconds.

Reply:
> This is probably fine, but it does seem like it could lead to a messier shutdown. I liked the idea of waiting to acquire the lock, but you're right that we don't want to wait forever. Maybe we could instead switch the acquire to take a timeout argument (that just goes onto the blpop) and then set a timeout here for a couple of seconds.

Why is this messier? It seems to work just fine and does the right thing: every worker tries to close and gives up if locked.
I had thought of the timeout option, but I don't like it much. Just because it takes longer.
But I think you are right, that just closing the lock without acquiring it is the same as what it does now but simpler.

Reply:
Yeah. I guess you are right on that one: if it can't acquire the lock it doesn't do anything. I wasn't reading very carefully and though if it couldn't acquire the lock it was just removing everything anyway. Alright, ignore this one.

Reply:
Wait, I thought you were proposing to just remove everything without acquiring the lock first. That should do the same as what I have now, except it will also close a locked lock instead of leaving it there...